pallets/transaction-multi-payment/src/lib.rs

```diff
@@ -453,13 +458,20 @@ where
 		let price = Pallet::<T>::get_currency_price(currency)
 			.ok_or(TransactionValidityError::Invalid(InvalidTransaction::Payment))?;
 
-		let converted_fee =
+		let mut converted_fee =
 			convert_fee_with_price(fee, price).ok_or(TransactionValidityError::Invalid(InvalidTransaction::Payment))?;
 
+		//Apply 10% discount on HDX
+		if currency == T::NativeAssetId::get() {
+			let fee_without_tip = converted_fee.saturating_sub(tip);
+			let fee_without_tip_discounted = NATIVE_FEE_DISCOUNT_MULTIPLIER.mul_floor(fee_without_tip);
+			converted_fee = fee_without_tip_discounted.saturating_add(tip);
```

Comment:
where did the tip go ? 

it seems that it is not withdrawn, and when it comes to correcting it , you add it back to corrected fee. 

although there seems to be a test to ensure issuance invariant - according to the code - i would say, it mints the tip ?! 

i would suggest to investigate bit further.

Reply:
i would suggest to add a simple test and call only pre_dispatch to see what happens with the tip?!

Reply:
yeah, we have to make sure the treasury receives tip and that it is not burned @dmoka